lib/JabberBot.js

```diff
@@ -0,0 +1,232 @@
+var Botkit = require(__dirname + '/CoreBot.js');
+const Stanza = require('node-xmpp-client').Stanza;
+const GroupManager = require('./JabberGroupManager.js')
+
+function JabberBot(configuration) {
+    // Create a core botkit bot
+    var controller = Botkit(configuration || {});
+
+
+    controller.middleware.format.use(function (bot, message, platform_message, next) {
+        // clone the incoming message
+        for (var k in message) {
+            platform_message[k] = message[k];
+        }
+        next();
+    });
+
+    // customize the bot definition, which will be used when new connections
+    // spawn!
+    controller.defineBot(function (botkit, config) {
+        var xmpp = require('simple-xmpp');
+
+        var bot = {
+            type: 'xmpp',
+            botkit: botkit,
+            config: config || {},
+            client_jid: config.client.jid,
+            utterances: botkit.utterances,
+        };
+
+        GroupManager(config, xmpp, bot, controller);
+
+        xmpp.on('online', function (data) {
+            console.log('Connected with JID: ' + data.jid.user);
+            console.log('Yes, I\'m connected!');
+
+        });
+
+        xmpp.on('close', function () {
+            console.log('connection has been closed!');
+        });
+
+        xmpp.on('error', function (err) {
+            console.log(err);
+        });
+
+        xmpp.on('subscribe', function (from) {
+            xmpp.acceptSubscription(from);
+            console.log('accept subscribe from' + from);
+            controller.trigger('subscribe', [bot, from]);
+        });
+
+        xmpp.on('unsubscribe', function (from) {
+            xmpp.acceptUnsubscription(from);
+        });
+
+        function findBotJid(jid) {
+            return jid === bot.client_jid;
+        }
+
+        function IsBotMentioned(message)
+        {
+            let mention_jids = ExtractMentionJids(message);
+            if (mention_jids.find(findBotJid))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        function ExtractMentionJids(message) {
```

Comment:
`ExtractMentionJids` function is not a constructor so it should start with lowercase `e`.